### PR TITLE
Minor doc enhancements to Security Basic Auth docs

### PR DESCRIPTION
**Pull request by michelle-purcell**

### CODE REVIEW

File: docs/src/main/asciidoc/security-identity-providers.adoc [hunk 1]
```diff
@@ -9,18 +8,18 @@ include::_attributes.adoc[]
 :diataxis-type: concept
 :categories: security
 
-In the Quarkus Security framework, identity providers play a key role in authentication and authorization, providing services for storing and verifying user identities.
-The Jakarta Persistence `IdentityProvider` creates a `SecurityIdentity` instance used during user authentication to verify and authorize access requests, making your Quarkus application secure.
+In the Quarkus Security framework, identity providers play a crucial role in authentication and authorization, providing services for storing and verifying user identities.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-identity-providers.adoc:11]: This text is modifief in another PR so there will be a merge conflict here

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc [hunk 2]
```diff
@@ -165,8 +168,8 @@ public class AdminResource {
     }
 }
 ----
-* Finally, implement the `/api/users/me` endpoint. As you can see from the source code example below, we are trusting only users with the `user` role.
-We are also using `SecurityContext` to get access to the currently authenticated `Principal`, and we return the username, all of which is loaded from the database.
+* Finally, implement an `/api/users/me` endpoint that trusts users who have the `user` role.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:171]:
```suggestion
* Finally, implement an `/api/users/me` endpoint that can only accessed by users who have the `user` role.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:171]: It does not sound quite right,  the endpoints themselves are not involved in the RBAC process

**Comment by sberyozkin** [comment]
Thanks Michelle, LGTM, minor comments added

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc (hunk 2 above)
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc]: @sberyozkin - Fixed. Thanks for your suggestion.

File: docs/src/main/asciidoc/security-identity-providers.adoc (hunk 1 above)
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-identity-providers.adoc]: @sberyozkin - Self-inflicted conflict fixed. The identity providers doc updates over on my other branch are now out of this branch and this is rebased to main. Thanks for flagging. I'll double-check this in the am with fresh eyes.

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**